sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/AesCryptographyProvider.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Security.KeyVault.Keys.Cryptography
+{
+    internal class AesCryptographyProvider : ICryptographyProvider
+    {
+        private readonly JsonWebKey _jwk;
+
+        internal AesCryptographyProvider(JsonWebKey jwk)
+        {
+            _jwk = jwk;
+        }
+
+        public bool ShouldRemote => _jwk.KeyId != null;
+
+        public bool SupportsOperation(KeyOperation operation)
+        {
+            if (_jwk != null)
+            {
+                if (operation == KeyOperation.Encrypt || operation == KeyOperation.Decrypt || operation == KeyOperation.WrapKey || operation == KeyOperation.UnwrapKey)
```

Comment:
I wonder if for now, for optimization, we remove the encrypt and decrypt implied support.